apps/sms/test/marionette/lib/messages.js

```diff
@@ -208,11 +208,9 @@ var ConversationAccessor = require('./views/conversation/accessors');
         },
 
         performHeaderAction: function() {
-          this.Composer.header.scriptWith(function(header) {
-            var event = document.createEvent('HTMLEvents');
-            event.initEvent('action', true, true);
-            header.dispatchEvent(event);
-          });
+          this.client.switchToShadowRoot(this.Composer.header);
```

Comment:
First of all, thanks for you contribution and interest in this bug! Please, see my comments below:

* Similar change should be made in 3 places: ```performHeaderAction```, ```performReportHeaderAction```, ```performGroupHeaderAction``` (everywhere inside marionette folder where we have workaround with ```'HTMLEvents'```);

* For this to work you should have ```headerActionButton``` field defined first, we don't have it yet :) So for this particular case you need to add action button selector to [1] and expose it via property, just see how it's done for the rest of properties. For ```performReportHeaderAction``` and ```performGroupHeaderAction```, you need to do the similar thing, but all the work will be done inside this file.

* Just push your changes into the same PR and please update commit message to contain bug description like: ```Bug 1206093 - [Messages][Tests] Leverage "client.switchToShadowRoot" to tap on gaia-header action button. r=azasypkin```.

* I'm not sure why integration tests didn't run automatically when you created PR, but once you update PR with your changes, ```Autolander``` should add special link your PR. This link will lead you to the integration test results. And once all tests are green, you can ask for review again :)

[1] https://github.com/mozilla-b2g/gaia/blob/899860aa2f58cb2597405cd3f8572a938e7ca384/apps/sms/test/marionette/lib/views/new-message/accessors.js#L17